src/w_wad.c

```diff
@@ -151,6 +151,17 @@ static void AddDirs(w_module_t *module, w_handle_t handle, const char *base)
     }
 }
 
+boolean W_ExtrasWad(const char *path)
+{
+    w_handle_t handle = {0};
+    if (w_file_module.Open(path, &handle) == W_FILE)
+    {
+        array_clear(wadfiles); // don't register it before IWAD
```

Comment:
This screams "bug!" all over the place. I don't even understand what happens here.

Reply:
The actual lump table is in `lumpinfo`. `wadfiles` is just a list of WAD names, for `wadfiles[0]` checks or save/autoload directories, not that important.

In `w_file.Open()` the name of the WAD is recorded in the `wadfiles` array, so we delete it and add it back after IWAD, so that other checks work. Ugly hack, but if we decide to implement an in-game launcher, we will have to redo this system anyway to load WADs on the fly.